app/models/ability.rb

```diff
@@ -68,6 +68,14 @@ def initialize(user)
         c.submissions.exists?(user_id: user.id, reviewed: true)
       end
 
+      cannot :read, Certificate
+      can :read, Certificate do |c|
+        c.user == user
+      end
+      can :create, Certificate do |c|
+        c.course.certificate_downloadable_for? user
```

Comment:
canfusing :create and then we check if it is downloadable. So are we creating the certs each time?